integration/container/stop_test.go

```diff
@@ -0,0 +1,74 @@
+package container
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/network"
+	"github.com/docker/docker/client"
+	"github.com/docker/docker/integration/util/request"
+	"github.com/gotestyourself/gotestyourself/icmd"
+	"github.com/gotestyourself/gotestyourself/poll"
+	"github.com/gotestyourself/gotestyourself/skip"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDeleteDevicemapper(t *testing.T) {
+	skip.IfCondition(t, testEnv.DaemonInfo.Driver != "devicemapper")
+
+	defer setupTest(t)()
+	client := request.NewAPIClient(t)
+	ctx := context.Background()
+
+	foo, err := client.ContainerCreate(ctx,
```

Comment:
Does this change actually need a container to exist, or could this be a unit test in `pkg/devicemapper` ?

Reply:
@dnephin The PR tries to fix the issue where an out-of-sync deletion of pool device will not prevent container deletion from returning an error. Though I am not very familiar with this part of the code. There is no existing unit test in `pkg/devicemapper` so not sure if it is possible to created a unit test easily.